pkg/test/golden/validator.go

```diff
@@ -168,13 +173,41 @@ func (v *validator) Validate(r declarative.Reconciler) {
 			continue
 		}
 
+		// Ignore any files that are not input - we want to find .side_in and .out files
+		// that correspond to given .in file
 		if !strings.HasSuffix(p, ".in.yaml") {
-			if !strings.HasSuffix(p, ".out.yaml") {
-				t.Errorf("unexpected file in tests directory: %s", p)
-			}
 			continue
 		}
 
+		objectsToCleanup := []runtime.Object{}
+		// Check if there is a file containing side inputs for this test
+		sideInputPath := strings.Replace(p, ".in.yaml", ".side_in.yaml", -1)
+		sideInputRead, err := os.ReadFile(sideInputPath)
+		if err != nil {
+			t.Logf("Could not read side input file %s: %v, skipping", sideInputPath, err)
```

Comment:
Nit: I like to check that the error is indeed os.IsNotFound(err) (I'm assuming that's what we're expecting), and avoid the log message - and possibly avoid covering up another error (like e.g. a permissions error)